File: src/main/java/com/nicholasnassar/sfsapi/data/links/AssignmentLink.java
package com.nicholasnassar.sfsapi.data.links;

import io.vertx.core.json.JsonObject;

public class AssignmentLink extends Link {
    private final String id;

    public AssignmentLink(String id) {
        super(LinkType.ASSIGNMENT);

        this.id = id;
    }

    @Override
    public JsonObject asJson() {
        return new JsonObject().put("type", type.getName()).put("id", id);
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/data/links/AssignmentLink.kt
package com.nicholasnassar.sfsapi.data.links

import io.vertx.core.json.JsonObject

class AssignmentLink(val id: String) : Link(LinkType.ASSIGNMENT) {
    override fun asJson(): JsonObject {
        return JsonObject().put("type", type.dataName).put("id", id)
    }
}
